packages/components/src/ResourcePicker/Resource/Resource.component.js

```diff
@@ -59,22 +87,20 @@ function Resource({ parent, index, style, className, t }) {
 			{icon && <Icon name={icon} />}
 			<div className={classNames('data-container', theme['data-container'])}>
 				<span className={classNames('title', theme.title)}>{name}</span>
-				{author ? (
-					<small className={classNames('author', theme.author)}>
-						{getAuthorLabel(t, author, modified)}
-					</small>
-				) : null}
-			</div>
-			<div className={classNames('flags-container', theme['flags-container'])}>
-				{Object.keys(FLAGS).map(flag => (
-					<Icon
-						className={classNames(theme.flag, {
-							[theme.visible]: flags.includes(flag),
-						})}
-						name={FLAGS[flag]}
-					/>
-				))}
+				<Subtitle author={author} modified={modified} subtitle={subtitle} t={t} />
 			</div>
+			{!isEmpty(flags) ? (
+				<div className={classNames('flags-container', theme['flags-container'])}>
+					{Object.keys(FLAGS).map(flag => (
+						<Icon
+							className={classNames(theme.flag, {
+								[theme.visible]: flags.includes(flag),
```

Comment:
my2cents: O(nm) wheras O(n) would be enough if flags was an object instead of an array